source/server/hot_restart_impl.cc

```diff
@@ -141,14 +141,16 @@ HotRestartImpl::HotRestartImpl(Options& options)
   RELEASE_ASSERT(rc != -1, "");
 }
 
-Stats::RawStatData* HotRestartImpl::alloc(const std::string& name) {
+Stats::RawStatData* HotRestartImpl::alloc(absl::string_view name) {
   // Try to find the existing slot in shared memory, otherwise allocate a new one.
   Thread::LockGuard lock(stat_lock_);
-  absl::string_view key = name;
-  if (key.size() > options_.statsOptions().maxNameLength()) {
-    key.remove_suffix(key.size() - options_.statsOptions().maxNameLength());
+  // The name is truncated in ThreadLocalStore before this is called.
+  size_t max_length = options_.statsOptions().maxNameLength();
+  if (name.length() > max_length) {
```

Comment:
Can this happen then? Should this just be an assert? The comment is a bit confusing.

Reply:
It should not happen in production. It can happen in a unit test.  It could be an ASSERT, but then it can't be tested.  It could be a RELEASE_ASSERT, and be tested with a DEATH_TEST.

I'm happy to go with whatever is the best practice of those for the project.

I'll try to clarify the comment.

Reply:
I would probably just do a normal ASSERT as this seems like a programming error. If you feel strongly that it should be a RELEASE_ASSERT w/ a test that's fine also.